web/src/components/TimeSliderInput.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import styled from 'styled-components';
+import { sortBy } from 'lodash';
+
+const COLORS = {
+  DAY: '#EFEFEF',
+  NIGHT: 'rgba(0, 0, 0, 0.15)',
+};
+
+const createGradient = sets => sortBy(sets, 'end')
+  .map(
+    ({ start, end }) => `${COLORS.DAY} ${start}%, ${COLORS.NIGHT} ${start}%, ${COLORS.NIGHT} ${end}%, ${COLORS.DAY} ${end}%`,
+  )
+  .join(',\n');
+
+export const StyledInput = styled.input`
+  --time-gradient: linear-gradient(
+    90deg,
+    ${props => createGradient(props.nightSets)}
+  );
+`;
+
+
+export const TimeSliderInput = ({
+  onChange, value, nightTimeSets, isValueAtNight, min, max,
+}) => {
+  const timeClass = isValueAtNight ? 'night' : 'day';
+  return (
+    <StyledInput
+      type="range"
+      className={`time-slider-input ${timeClass}`}
+      nightSets={nightTimeSets}
```

Comment:
Also curious, when looking at your preview, it does not seem like there are any gradients, so what's their use here ?

Reply:
The gradient is what sets the gray colors on the bar (the steps are "hard", so there's no fluid transition between the colors). It's a (kinda hacky) way to change the background color of an input field without having multiple elements :)